models/registro_plan_adquisicion.go

```diff
@@ -152,25 +153,186 @@ func BuscarIndexPorCampo(RegistroPlanAdquisicion []map[string]interface{}, Rubro
-			return nil, errorDatos
+		resultadoPlan, err := IngresoRenglonPlanFuncionamiento(registroPlanAdquisicion)
+		if err != nil {
+			logs.Error(err)
+			outputError = map[string]interface{}{
+				"funcion": "IngresoPlanAdquisicion -  IngresoRenglonPlanFuncionamiento(registroPlanAdquisicion)",
+				"err":     err,
+				"status":  "502",
+			}
+			return nil, outputError
 		} else {
-			return result, nil
+			idPlanAdquisiciones := int(registroPlanAdquisicion["PlanAdquisicionesId"].(float64))
+			if resultado, err := ObtenerMovimientosProcesoExternoPlan(idPlanAdquisiciones); err != nil {
+				// DATOS DE PRUEBA
+				/*resultadoPrueba := []MovimientoProcesoExternoId{
+					{
+						Id: 15,
+					},
+				}
+				var errorPrueba interface{}
+				errorPrueba = nil
+				if resultado, errorDatos := resultadoPrueba, errorPrueba; errorDatos != nil {*/
+				logs.Error(err)
+				outputError = map[string]interface{}{
+					"funcion": "IngresoPlanAdquisicion -  ObtenerMovimientosProcesoExternoPlan(idPlanAdquisiciones)",
+					"err":     err,
+					"status":  "502",
+				}
+				return nil, outputError
+			} else {
+				if len(resultado) > 0 {
+					logs.Debug(resultado)
+					movimientoExternoID := resultado[0].Id
+					if resultado, err := ObtenerRegistroMovimientoFuncionamiento(registroPlanAdquisicion, movimientoExternoID); err != nil {
+						logs.Error(err)
+						outputError = map[string]interface{}{
+							"funcion": "IngresoPlanAdquisicion -  ObtenerMovimientosProcesoExternoPlan(idPlanAdquisiciones)",
+							"err":     err,
+							"status":  "502",
+						}
+						return nil, outputError
+					} else {
+						logs.Debug(resultado)
+						return resultadoPlan, nil
+					}
+				} else {
+					if resultado, err := ObtenerMovimientoProcesoExterno(registroPlanAdquisicion); err != nil {
+						logs.Error(err)
+						outputError = map[string]interface{}{
+							"funcion": "IngresoPlanAdquisicion -  ObtenerMovimientosProcesoExternoPlan(idPlanAdquisiciones)",
+							"err":     err,
+							"status":  "502",
+						}
+						return nil, outputError
+					} else {
+						logs.Debug(resultado)
+						if resultado, err := ObtenerMovimientosProcesoExternoPlan(idPlanAdquisiciones); err != nil {
+							logs.Error(err)
+							outputError = map[string]interface{}{
+								"funcion": "IngresoPlanAdquisicion -  ObtenerMovimientosProcesoExternoPlan(idPlanAdquisiciones)",
+								"err":     err,
+								"status":  "502",
+							}
+							return nil, outputError
+						} else {
+							logs.Debug(resultado)
+							movimientoExternoID := resultado[0].Id
+							if resultado, err := ObtenerRegistroMovimientoFuncionamiento(registroPlanAdquisicion, movimientoExternoID); err != nil {
+								logs.Error(err)
+								outputError = map[string]interface{}{
+									"funcion": "IngresoPlanAdquisicion -  ObtenerMovimientosProcesoExternoPlan(idPlanAdquisiciones)",
+									"err":     err,
+									"status":  "502",
+								}
+								return nil, outputError
```

Comment:
Si `ObtenerRegistroMovimientoFuncionamiento` ...
```suggestion
								return nil, err
```